### PR TITLE
New build process

### LINKED ISSUE #39: Add the new release process

### PR DESCRIPTION
**Pull request by discordianfish**
This closes #39

### CODE REVIEW

**Comment by juliusv** [comment]
@sdurrheimer

File: .promu.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+go: 1.5.4
```
Comment (sdurrheimer) [inline .promu.yml]: Given that Prometheus and Alertmanager are now `1.6.2`, I think we can change here too.

File: .promu.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+crossbuild:
+    platforms:
+        - linux/amd64
+        - linux/386
+        - darwin/amd64
+        - darwin/386
+        - windows/amd64
+        - windows/386
+        - netbsd/amd64
+        - netbsd/386
```
Comment (sdurrheimer) [inline .promu.yml]: Are these the full working platforms for `statsd_exporter` ?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+sudo: false
+
+language: go
+go:
+- 1.5
```
Comment (sdurrheimer) [inline .travis.yml]: `1.5.4` and `1.6.2` would be better.

File: Dockerfile [hunk 4]
```diff
@@ -1,4 +1,7 @@
-FROM        sdurrheimer/alpine-golang-make-onbuild
+FROM        quay.io/prometheus/busybox:glibc
```
Comment (sdurrheimer) [inline Dockerfile]: Is the `linux/amd64` binary for `statsd_exporter` not statically buildable ?

File: .promu.yml (hunk 1 above)
Comment (discordianfish) [inline .promu.yml]: Fixed!

File: README.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
-statsd_exporter
-=============
+# statsd exporter [![Build Status](https://travis-ci.org/prometheus/statsd_exporter.svg)][travis]
+
+[![CircleCI](https://circleci.com/gh/prometheus/statsd_exporter/tree/master.svg?style=shield)][circleci]
+[![Image Size & Layers](https://imagelayers.io/badge/prom/statsd-exporter:master.svg)][imagelayers]
+[![Docker Stars](https://img.shields.io/docker/stars/prom/statsd-exporter.svg)][hub]
+[![Docker Pulls](https://img.shields.io/docker/pulls/prom/statsd-exporter.svg)][hub]
```
Comment (sdurrheimer) [inline README.md]: The shields discussion is [still going](https://github.com/prometheus/prometheus/pull/1598), but I think we should replace the `imagelayer` shield by a Quay link and remove the `Docker Stars`.

```
[![Docker Repository on Quay](https://quay.io/repository/prometheus/statsd-exporter/status)][quay]

[quay]: https://quay.io/repository/prometheus/statsd-exporter
```

File: .travis.yml (hunk 3 above)
Comment (discordianfish) [inline .travis.yml]: Both? Or just 1.6.2? I simply used the node-exporter as example. I'll change it for now to 1.6.2.

Comment (discordianfish) [inline .travis.yml]: Just tested it and they can all be built. Do we want to support more than those?

File: Dockerfile (hunk 4 above)
Comment (discordianfish) [inline Dockerfile]: Changed it to busybox:latest.

File: .travis.yml (hunk 3 above)
Comment (sdurrheimer) [inline .travis.yml]: People might still build with `go 1.5`, so I would build with `1.5.4` and `1.6.2`.
I don't really have a opinion about `tip`, is a `devel` version really useful ?

File: .promu.yml (hunk 2 above)
Comment (discordianfish) [inline .promu.yml]: Took that from node-exporter. I've tried and they all can be built. Do we want to support more/less platforms?

Comment (sdurrheimer) [inline .promu.yml]: Given that there a lot of release, we can do the full list if they all works:

```
    platforms:
        - linux/amd64
        - linux/386
        - darwin/amd64
        - darwin/386
        - windows/amd64
        - windows/386
        - freebsd/amd64
        - freebsd/386
        - openbsd/amd64
        - openbsd/386
        - netbsd/amd64
        - netbsd/386
        - dragonfly/amd64
        - linux/arm
        - linux/arm64
        - freebsd/arm
        - openbsd/arm
        - netbsd/arm
        - linux/ppc64
        - linux/ppc64le
        - linux/mips64
        - linux/mips64le
```

Comment (grobie) [inline .promu.yml]: If we support all platforms, let's not specify it here in promu and use the defualt. then we can add new versions without the need to touch x repositories.

Comment (sdurrheimer) [inline .promu.yml]: Good point 👍

**Comment by sdurrheimer** [comment]
👍

**Pull request merged**